pyspectre/core.py

```diff
@@ -138,14 +142,13 @@ def start_session( net_path: str, includes: Iterable[str] = None
     log    = log_fifo(os.path.splitext(raw)[0])
     inc    = [] if not includes else [f'-I{os.path.expanduser(i)}' for i in includes]
     cmd    = 'spectre'
-    args   = ['-64', '+interactive', '-format nutbin', f'-raw {raw}'
-             , f'=log {log}' ] + inc + [net]
+    args   = ['-64', '+interactive', '-format', 'nutbin', '-raw', f'{raw}', f'+log', f'{log}'] + inc + [net]
 
     if not os.path.isfile(net):
         raise(FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), net))
     if not os.access(net, os.R_OK):
         raise(PermissionError(errno.EACCES, os.strerror(errno.EACCES), net))
-
+    # Spectre interactive session host handle ?
```

Comment:
expect launches an interactive process of any kind, `spectre` in this case, and let's us interact with it programmatically. So yes, `repl` in this case is just what you would see when launching `spectre +interactive` and we can read from and write to it with plain text as we would using a terminal and keyboard.

Reply:
If you've ever used ngspice interactively, it works similar to that. When running just a single simulation with spectre, it starts by compiling all the models from the PDK and it has to this every time it is launched. The idea with interactive mode is to do this step just once. When you start a session you open a spectre REPL with a netlist and have access to all `parameters` and analyses defined in this netlist. With the `scl*` commands we can adjust all the parameters and run all the simulations for as long as the session, or repl, is open. The intended use is to start a session once and then keep it open and iteratively run as many simulations as needed. [serafin](https://github.com/AugustUnderground/serafin) demonstrates this in conjunction with multiprocessing, launching a number of sessions and simulating in parallel. The session stays open for as long as the object is in scope. As long as your program doesn't terminate or the garbage collector `del`s the object, the session and all it's parameters are accessible. The simulation results are stored on disk as raw files and can be available afterwards. As I've mentioned above, we're using this as backend for and RL env and also for data generation, so long run times are our main use case.